### PR TITLE
Avoid exceptions in tearDown of io and perf tests when canceling

### PR DESCRIPTION
**Pull request by dsilakov**
Now, depending on the test, we either init the variables used in tearDown before
the test cancel or check if we have necessary attribute before using it in tearDown.

Signed-off-by: Denis Silakov <dsilakov@gmail.com>

### CODE REVIEW

**Review by PraveenPenguin — APPROVED** [review]
Looks Good to me @Naresh-ibm can you please have a look on this as well

File: io/driver/driver_parameter.py [hunk 1]
```diff
@@ -157,9 +159,10 @@ def tearDown(self):
...
         """
         self.log.info("Restoring Default param")
-        linux_modules.unload_module(self.module)
-        linux_modules.load_module(self.module)
-        time.sleep(self.load_unload_sleep_time)
-        if linux_modules.module_is_loaded(self.module) is False:
-            self.fail("Cannot restore default values for Module : %s"
-                      % self.module)
+        if self.module:
+            linux_modules.unload_module(self.module)
```
[the remark this answers is not in the document]
Comment (dsilakov) [reply io/driver/driver_parameter.py:163]: I wonder why would you need this sub-module handling in the tests.  linux_modules.unload_module in avocado already takes care about submodules: https://github.com/avocado-framework/avocado/blob/master/avocado/utils/linux_modules.py#L174

**Comment by abdhaleegit** [comment]
@dsilakov multiple changes in single commit, would be good to segregate commits for each different logical change

**Comment by PraveenPenguin** [comment]
> @dsilakov multiple changes in single commit, would be good to segregate commits for each different logical change

SIngle commit for this Fix should look good , @Naresh-ibm can you please have look

**Comment by dsilakov** [comment]
Not a big problem, split into smaller commits.

**Comment by PraveenPenguin** [comment]
@Naresh-ibm review please

**Review by Naresh-ibm — APPROVED** [review]

**Pull request merged**